src/components/dialogs/ConfirmDialog.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <v-dialog
+    v-model="dialog"
+    content-class="confirm-dialog"
+    :max-width="options.width"
+    :style="{ zIndex: options.zIndex }"
+    :persistent="options.persistent"
+    @keydown.esc="onClickCancel">
+
+    <v-card>
+      <v-card-title>{{ title }}</v-card-title>
+      <v-card-text class="black--text" v-show="!!message" v-html="message"></v-card-text>
+      <v-divider class="my-0" v-show="!!options.yes || !!options.no || !!options.cancel"></v-divider>
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn id="dialog-yes-button" color="primary" text v-show="!!options.yes"
+          @click.native="onClickYes()">{{ options.yes }}</v-btn>
+        <v-btn id="dialog-no-button" color="primary" text v-show="!!options.no"
+          @click.native="onClickNo()">{{ options.no }}</v-btn>
+        <v-btn id="dialog-cancel-button" color="secondary" text v-show="!!options.cancel"
+          @click.native="onClickCancel()">{{ options.cancel }}</v-btn>
+      </v-card-actions>
+    </v-card>
+
+  </v-dialog>
+</template>
+
+<script lang="ts">
+/**
+ * Vuetify Confirm Dialog component
+ * ref: https://gist.github.com/eolant/ba0f8a5c9135d1a146e1db575276177d
+ *
+ * Insert component where you want to use it:
+ * <confirm ref="confirm"></confirm>
+ *
+ * Call it using promise:
+ * this.$refs.confirm.open('Delete', 'Are you sure?', { color: 'red' }).then((confirm) => {})
+ *
+ * Or use await:
+ * if (await this.$refs.confirm.open('Delete', 'Are you sure?', { color: 'red' })) {
+ *   // yes
+ * }
+ * else {
+ *   // cancel
+ * }
+ *
+ * Alternatively you can place it in main App component and access it globally via this.$root.$confirm
+ * <template>
+ *   <v-app>
+ *     ...
+ *     <confirm ref="confirm"></confirm>
+ *   </v-app>
+ * </template>
+ *
+ * mounted() {
+ *   this.$root.$confirm = this.$refs.confirm.open
+ * }
+ */
```

Comment:
Was this a quick reference for development?

Reply:
Yes, this is doc.

I think this is the exact same component used in business-filings-ui.

Reply:
Oh interesting, thanks!